src/openloco/windows/stationwnd.cpp

```diff
@@ -1,55 +1,660 @@
+    namespace cargo
     {
-        registers regs;
-        regs.al = amount;
-        regs.cx = x;
-        regs.dx = y;
-        regs.esi = (int32_t)&w;
-        regs.edi = (int32_t)&dpi;
-        regs.ebp = colour;
-        call(0x0048EF02, regs);
+        enum widx
+        {
+            scrollview = 7,
+            status_bar,
+            station_catchment,
+        };
+
+        static widget_t widgets[] = {
+            commonWidgets(223, 136),
+            make_widget({ 3, 44 }, { 219, 80 }, widget_type::scrollview, 1, 2),
+            make_widget({ 3, 125 }, { 195, 10 }, widget_type::wt_13, 1),
+            make_widget({ 198, 44 }, { 24, 24 }, widget_type::wt_9, 1, image_ids::show_station_catchment, string_ids::station_catchment),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << station_catchment);
+
+        // 0x0048E7C0
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            self->widgets[widx::scrollview].right = self->width - 24;
+            self->widgets[widx::scrollview].bottom = self->height - 14;
+
+            self->widgets[widx::status_bar].top = self->height - 12;
+            self->widgets[widx::status_bar].bottom = self->height - 3;
+            self->widgets[widx::status_bar].right = self->width - 14;
+
+            self->widgets[widx::station_catchment].right = self->width - 2;
+            self->widgets[widx::station_catchment].left = self->width - 25;
+
+            common::repositionTabs(self);
+        }
+
+        // 0x0048E8DE
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            auto buffer = const_cast<char*>(stringmgr::get_string(string_ids::buffer_1250));
+            buffer = stringmgr::format_string(buffer, string_ids::accepted_cargo_separator);
+
+            auto station = stationmgr::get(self->number);
+            uint8_t cargoTypeCount = 0;
+
+            for (uint32_t cargoId = 0; cargoId < max_cargo_stats; cargoId++)
+            {
+                auto& stats = station->cargo_stats[cargoId];
+
+                if (!stats.is_accepted())
+                    continue;
+
+                *buffer++ = ' ';
+                *buffer++ = control_codes::inline_sprite_str;
+                *(reinterpret_cast<uint32_t*>(buffer)) = objectmgr::get<cargo_object>(cargoId)->unit_inline_sprite;
+                buffer += 4;
+
+                cargoTypeCount++;
+            }
+
+            if (cargoTypeCount == 0)
+            {
+                buffer = stringmgr::format_string(buffer, string_ids::cargo_nothing_accepted);
+            }
+
+            *buffer++ = '\0';
+            auto args = FormatArguments();
+            args.push(string_ids::buffer_1250);
```

Comment:
I don't think this is required. We don't use the format args by my reading of this function.